### PR TITLE
Eliminate online gas estimation from send and invite flows

### PR DESCRIPTION
**Pull request by nategraf**
### Description

As described in https://github.com/celo-org/celo-monorepo/issues/5432, gas estimation is a costly
operation for the mobile client. The most promising short-term solution is to simply hard-code the
gas value to be used in for simple transactions along common code paths. The most common transaction
sent by a typical user is to send cUSD, either by a direct transfer or an invite.

This PR adds the required plumbing to use a single gas value, price and currency throughout the send
and invite flows. As a result, a single constant gas value can be used from start to finish. The
major result of this is that instead of estimating gas online twice, as the app currently does, the
static value is used and online gas estimation is never used.

This represents a large improvement to the experienced transaction speed, especially in less than
ideal network conditions. Additional effects of this are that errors caused by gas estimation should
be reduced, it's possible to give a fee estimate while offline, and the fee estimate shown the user
will be respected as a maximum fee value.

### Other changes

* Remove the unused `EstimateFee.tsx`
* Refactor cancellation for `sendTransaction`
* Fix gas price cache to handle multiple currencies.
* Propagate receipts back up from `sendTransaction`
* Use a Lock for getting contractkit instance to avoid reliance on polling and flag-based locking.
* Don't clear the transaction feed when the call to the blockchain API fails, such as when the device is offline.
* Add `networkConnectedSelector` to access the network connectivity state reported by the device.

### Tested

* Updated snapshot and unit tests to cover new functionality
* Manually tests effected user flows: Sending cUSD, escrow, and reclaiming escow

### Related issues

- Related: https://github.com/celo-org/celo-monorepo/issues/5432

### Backwards compatibility

No backwards compatibility concerns.

### CODE REVIEW

**Comment by nategraf** [comment]
Just noticed in the logs the following message.

```
[Fri Mar 12 2021 17:09:41.193]  INFO     Warning: Cannot update a component from inside the function body of a different component.%s ["
    in CalculateSendFee (at CalculateFee.tsx:138)
    in CalculateFee (at SendConfirmation.tsx:451)
    in SendConfirmation (at SceneView.tsx:122)
    in StaticContainer
```
I have a feeling it's related to this line below, but I'm not exactly sure what to do differently. Maybe a helpful reviewer could suggest a fix!
https://github.com/celo-org/wallet/blob/ecfa456aa4e2ba7a4b102b4b58a6aa196016b945/packages/mobile/src/send/SendConfirmation.tsx#L244

**Comment by etuleu** [comment]
> Warning: Cannot update a component from inside the function body of a different component

Sounds like this: https://reactjs.org/blog/2020/02/26/react-v16.13.0.html#warnings-for-some-updates-during-render

Maybe we can fix these in a different PR though unless it's easy to make the change. It's not a huge problem for now, but we should fix these warnings at some point.

File: packages/mobile/src/send/SendConfirmation.tsx [hunk 1]
```diff
@@ -278,10 +297,13 @@ function SendConfirmation(props: Props) {
         dekFee = fee.dividedBy(2)
       }
 
-      ValoraAnalytics.track(FeeEvents.fee_rendered, {
-        feeType: 'Security',
-        fee: securityFee ? securityFee.toString() : securityFee,
-      })
+      // DO NOT MERGE: Test this out.
```
Comment (etuleu) [inline packages/mobile/src/send/SendConfirmation.tsx:300]: reminder

Comment (nategraf) [reply packages/mobile/src/send/SendConfirmation.tsx]: Thanks! I went ahead and tested this manually and it seems to work as intended. (Not sure how I might add an automated test)

**Review by etuleu — APPROVED** [review]
I looked over the changes but for me at this point this is mostly a learning exercise. Would be good to get Jean's eyes over this as well for sure.

**Comment by Celoqa** [comment]
Hi @nategraf I have verified this issue using Android Internal Build V1.12.0(1004294340)  & observed following:
**Observations:** 

- **4G Network:** Observed that, app takes 25 sec to load sending screen with estimation after tapping on Send button. Also it is observed that more than one gas fees are displaying.
- **3G Network:** Observed that, app takes more that a minute to load estimation on sending screen. Also observed user multiple fees are applied.
- **512 kbps speed:(using Charles)** Observed that, continuous loading indicator displayed for longer time & user not able to proceed from Send screen after tapping on Review button.
**Verified on devices:** Samsung Galaxy A5(7.0), iPhone XR (14.2)

Let me know if anything else need to test.
Thanks..!!

**Pull request merged**